homeassistant/components/met/__init__.py

```diff
@@ -159,15 +157,6 @@ def set_coordinates(self) -> bool:
             longitude = self._config[CONF_LONGITUDE]
             elevation = self._config[CONF_ELEVATION]
 
-        if not self._is_metric:
-            elevation = int(
-                round(
-                    DistanceConverter.convert(
-                        elevation, UnitOfLength.FEET, UnitOfLength.METERS
-                    )
-                )
-            )
```

Comment:
This is breaking existing configurations?

Reply:
Per #84895, the existing unit conversion may be incorrect, since Core always stores elevation in meters regardless of which unit? Also, I believe `met.no` expects elevation in meters. 

Any suggestions on how we can handle this?

Reply:
> since Core always stores elevation in meters regardless of which unit?

Core does, but config entries don't. Which is the part that is now breaking.

`elevation = self._config[CONF_ELEVATION]` <- data from config entry, not core.

../Frenck